File: app/src/main/java/com/cliniva/enventory/viewholder/AvailableProductHolder.java
package com.cliniva.enventory.viewholder;

/* Created by Imran Khan on 20-Jun-19.
 * Copyright (c) Imran Khan All rights reserved.*/

import android.databinding.ViewDataBinding;
import android.support.annotation.NonNull;

import com.cliniva.enventory.adapter.base.BaseRecyclerClickListener;
import com.cliniva.enventory.adapter.base.BaseRecyclerViewHolder;
import com.cliniva.enventory.databinding.ItemAddProductBinding;
import com.cliniva.enventory.databinding.ItemAvailableProductBinding;
import com.cliniva.enventory.model.AddProduct;
import com.cliniva.enventory.model.Product;

public class AvailableProductHolder extends BaseRecyclerViewHolder<Product, BaseRecyclerClickListener<Product>> {

    private ItemAvailableProductBinding binding;

    public AvailableProductHolder(@NonNull ViewDataBinding itemView) {
        super(itemView);
        this.binding = (ItemAvailableProductBinding) itemView;
    }

    @Override
    public void onBindView(Product item, int position, BaseRecyclerClickListener<Product> listener) {
        enableItemViewClick(item, listener);
        binding.setProduct(item);
    }
}

File: app/src/main/java/com/amirsons/inventory/recyclerview/viewholder/AvailableProductHolder.kt
package com.amirsons.inventory.recyclerview.viewholder

/* Created by Imran Khan on 20-Jun-19.
 * Copyright (c) Imran Khan All rights reserved.*/

import android.view.View

import com.amirsons.inventory.recyclerview.base.BaseRecyclerClickListener
import com.amirsons.inventory.recyclerview.base.BaseRecyclerViewHolder
import com.amirsons.inventory.datamanager.model.Product

class AvailableProductHolder(itemView: View) : BaseRecyclerViewHolder<Product, BaseRecyclerClickListener<Product>>(itemView) {

    override fun onBindView(item: Product, position: Int, listener: BaseRecyclerClickListener<Product>?) {
        enableItemViewClick(item, listener)
    }
}
